### DISCUSSION TITLE
Form is causing infinite loop

### DISCUSSION THREAD

**Discussion by GonzalesMatthew** [nss-evening-cohort-14/evening-client-side#281] [Help Me!] [answered]
Matthew Gonzales

I haven't seen anything related to this. I am encountering this strange error I don't know how to fix. After 6 hours of troubleshooting I finally narrowed it down to this error though: 

index.js:1 Warning: unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering.

A google search showing that this occurs when trying to update the DOM before the component has mounted 
(https://stackoverflow.com/questions/58040892/cannot-flush-updates-when-react-is-already-rendering)
-- however I don't know how to fix this. Like I said I've been troubleshooting this for 6 hours over the last few days. First I thought it was my modal, so just used a simple form, but now it is the way my form is being rendered? 

Whenever I open my form, which uses hooks to populate drop-down lists, it someone is closing and opening the modal again? I'm not sure, but the debugger statement I inserted above my hooks shows that my code keeps revisiting the Component again from the top. 

I'm going to continue troubleshooting on my own but could really use some help on how to resolve this.

I pushed my branch here and provided the link to my form. I'll be experimenting with getting rid of this error in the meantime.

https://github.com/GonzalesMatthew/CAPSTONE-matts-music-monitor/blob/submit-form-enabled-bug/src/components/forms/TaskForm.js

**Comment by GonzalesMatthew** [comment]
I further narrowed it down to my hooks. I commented them out and the errors are gone. Next, it isn't an infinite loop, just a compounding error as the code progresses. So it has something to do with my hooks. Going to investigate how what I'm doing is different from Chie's method because she is doing something similar for her form.

**Comment by TrinityChristiana** [comment, marked as answer]
Check your `useEffects` [here](https://github.com/GonzalesMatthew/CAPSTONE-matts-music-monitor/blob/submit-form-enabled-bug/src/components/forms/TaskForm.js#L48-L56). They are missing a second argument 😄

**Reply by GonzalesMatthew** [reply]
I'm going to go cry now. Can't believe I spent so much time on this. Thanks, Trin! :)
